lib/smack/Prelude.cpp

```diff
@@ -0,0 +1,438 @@
+#include "smack/Prelude.h"
+#include "smack/Naming.h"
+#include "smack/SmackOptions.h"
+#include "smack/Regions.h"
+
+namespace smack {
+
+const std::vector<unsigned> INTEGER_SIZES = {1, 5, 6, 8, 16, 24, 32, 40, 48,
+  56, 64, 80, 88, 96, 128, 160, 256};
+
+const std::vector<unsigned> REF_CONSTANTS = {
+  0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14,
+  1024
+};
+
+const std::vector<std::string> BINARY_OPS {"add", "sub", "mul", "sdiv",
+  "smod", "srem", "udiv", "urem", "shl", "lshr", "ashr", "and", "or", "xor",
+  "nand"};
+
+const std::vector<std::string> COMP_OPS {"ule", "ult", "uge", "ugt", "sle",
+  "slt", "sge", "sgt"};
+
+const std::vector<std::string> EQ_OPS {"eq", "ne"};
+
+const std::vector<BinExpr::Binary> BINARY_OPS_BODY {BinExpr::Plus,
+  BinExpr::Minus, BinExpr::Times};
+
+const std::vector<BinExpr::Binary> COMP_OPS_BODY {BinExpr::Lte, BinExpr::Lt,
+  BinExpr::Gte, BinExpr::Gt, BinExpr::Lte, BinExpr::Lt, BinExpr::Gte,
+  BinExpr::Gt};
+
+const std::vector<BinExpr::Binary> EQ_OPS_BODY {BinExpr::Eq, BinExpr::Neq};
+
+FuncDecl* Prelude::uninterpretedUnaryOp(int width, std::string name) {
+  std::string type = "i" + std::to_string(width);
+  return Decl::function(SmackRep::indexedName(name, {type}), {{"i1", type}},
```

Comment:
`"i1"` is like magic number in C. We can avoid it by using a function like `makeIntVar` or `makeIntVars` depending on the number of arguments. By this way, you can also make the argument names consistent in the function body.